src/metabase/models/database.clj

```diff
@@ -205,11 +205,8 @@
                            :existing-engine existing-engine
                            :new-engine      new-engine})))))))
 
-(defn- pre-insert [{:keys [details], :as database}]
-  (-> (cond-> database
-        (not details) (assoc :details {}))
-      handle-secrets-changes
-      (assoc :initial_sync_status "incomplete")))
+(defn- pre-insert [database]
+  (handle-secrets-changes (merge {:details {} :initial_sync_status "incomplete"} database)))
```

Comment:
Sorry for the long line, but I couldn't justify the `->` anymore 😁

Reply:
Except the necessary merge-with to make it actually work was too ugly, so back to cond-> for readability.